File: py/IEC61968/Assets/TestStandard.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106
from IEC61968.Assets.ASTMStandard import ASTMStandard
from IEC61968.Assets.CIGREStandard import CIGREStandard
from IEC61968.Assets.DINStandard import DINStandard
from IEC61968.Assets.DobleStandard import DobleStandard
from IEC61968.Assets.EPAStandard import EPAStandard
from IEC61968.Assets.IECStandard import IECStandard
from IEC61968.Assets.IEEEStandard import IEEEStandard
from IEC61968.Assets.ISOStandard import ISOStandard
from IEC61968.Assets.LaborelecStandard import LaborelecStandard
from IEC61968.Assets.TAPPIStandard import TAPPIStandard
from IEC61968.Assets.TestMethod import TestMethod
from IEC61968.Assets.TestVariantKind import TestVariantKind
from IEC61968.Assets.UKMinistryOfDefenceStandard import UKMinistryOfDefenceStandard
from IEC61968.Assets.WEPStandard import WEPStandard
from IEC61970.Base.Core.IdentifiedObject import IdentifiedObject


class TestStandard(IdentifiedObject):

    def __init__(self):
        super().__init__()
        self.test_method = TestMethod.D1275A
        self.test_standard_astm = ASTMStandard()
        self.test_standard_cigre = CIGREStandard()
        self.test_standard_din = DINStandard()
        self.test_standard_doble = DobleStandard()
        self.test_standard_epa = EPAStandard()
        self.test_standard_iec = IECStandard()
        self.test_standard_ieee = IEEEStandard()
        self.test_standard_iso = ISOStandard()
        self.test_standard_laborelec = LaborelecStandard()
        self.test_standard_tappi = TAPPIStandard()
        self.test_standard_uk_ministry_of_defence = UKMinistryOfDefenceStandard()
        self.test_standard_wep = WEPStandard()
        self.test_variant = TestVariantKind.TESTING_DONE_AT_TEMPERATURE_OF_0_C

File: java/IEC61968/Assets/TestStandard.java
package IEC61968.Assets;

import IEC61970.Base.Core.IdentifiedObject;

/**
 * The precise standard used in executing a lab test, including the standard, and
 * standard version, test method and variant, if needed.
 * @author ppbr003
 * @version 1.0
 * @created 25-Dec-2023 8:45:25 PM
 */
public class TestStandard extends IdentifiedObject {

	/**
	 * Identification of test method used if multiple methods specified by test
	 * standard.
	 */
	public TestMethod testMethod;
	/**
	 * Which ASTM standard used to determine analog value result. Applies only if ASTM
	 * standard used.
	 */
	public ASTMStandard testStandardASTM;
	/**
	 * Which CIGRE standard used to determine analog value result. Applies only if
	 * CIGRE standard used.
	 */
	public CIGREStandard testStandardCIGRE;
	/**
	 * Which DIN standard used to determine analog value result. Applies only if DIN
	 * standard used.
	 */
	public DINStandard testStandardDIN;
	/**
	 * Which Doble standard used to determine analog value result. Applies only if
	 * Doble standard used.
	 */
	public DobleStandard testStandardDoble;
	/**
	 * Which EPA standard used to determine analog value result. Applies only if EPA
	 * standard used.
	 */
	public EPAStandard testStandardEPA;
	/**
	 * Which IEC standard used to determine analog value result. Applies only if IEC
	 * standard used.
	 */
	public IECStandard testStandardIEC;
	/**
	 * Which IEEE standard used to determine analog value result. Applies only if IEEE
	 * standard used.
	 */
	public IEEEStandard testStandardIEEE;
	/**
	 * Which ISO standard used to determine analog value result. Applies only if ISO
	 * standard used.
	 */
	public ISOStandard testStandardISO;
	/**
	 * Which Laborelec standard used to determine analog value result. Applies only if
	 * Laborelec standard used.
	 */
	public LaborelecStandard testStandardLaborelec;
	/**
	 * Which TAPPI standard used to determine analog value result. Applies only if
	 * TAPPI standard used.
	 */
	public TAPPIStandard testStandardTAPPI;
	/**
	 * Which UK Ministry of Defence standard used to determine analog value result.
	 * Applies only if UK Ministry of Defence standard used.
	 */
	public UKMinistryOfDefenceStandard testStandardUKMinistryOfDefence;
	/**
	 * Which WEP standard used to determine analog value result. Applies only if WEP
	 * standard used.
	 */
	public WEPStandard testStandardWEP;
	/**
	 * Identification of variant of test method or standard if one is specified by the
	 * standard.
	 */
	public TestVariantKind testVariant;

	public TestStandard(){

	}
}//end TestStandard